File: java-sdk/lib/src/main/java/models/shared/WebLink.java
package .models.shared;

import java.time.OffsetDateTime;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import .utils.DateTimeSerializer;
import .utils.DateTimeDeserializer;

/**
 * WebLink
 * Web links are objects that point to URLs. These objects
 * are also known as bookmarks within the Box web application.
 * 
 * Web link objects are treated similarly to file objects,
 * they will also support most actions that apply to regular files.
**/
public class WebLink {
    @JsonInclude(Include.NON_ABSENT)
    @JsonSerialize(contentUsing = DateTimeSerializer.class)
    @JsonDeserialize(contentUsing = DateTimeDeserializer.class)
    @JsonProperty("created_at")
    public OffsetDateTime createdAt;
    public WebLink withCreatedAt(OffsetDateTime createdAt) {
        this.createdAt = createdAt;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("created_by")
    public UserMini createdBy;
    public WebLink withCreatedBy(UserMini createdBy) {
        this.createdBy = createdBy;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("description")
    public String description;
    public WebLink withDescription(String description) {
        this.description = description;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("etag")
    public String etag;
    public WebLink withEtag(String etag) {
        this.etag = etag;
        return this;
    }
    @JsonProperty("id")
    public String id;
    public WebLink withId(String id) {
        this.id = id;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("item_status")
    public WebLinkItemStatusEnum itemStatus;
    public WebLink withItemStatus(WebLinkItemStatusEnum itemStatus) {
        this.itemStatus = itemStatus;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonSerialize(contentUsing = DateTimeSerializer.class)
    @JsonDeserialize(contentUsing = DateTimeDeserializer.class)
    @JsonProperty("modified_at")
    public OffsetDateTime modifiedAt;
    public WebLink withModifiedAt(OffsetDateTime modifiedAt) {
        this.modifiedAt = modifiedAt;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("modified_by")
    public UserMini modifiedBy;
    public WebLink withModifiedBy(UserMini modifiedBy) {
        this.modifiedBy = modifiedBy;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("name")
    public String name;
    public WebLink withName(String name) {
        this.name = name;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("owned_by")
    public UserMini ownedBy;
    public WebLink withOwnedBy(UserMini ownedBy) {
        this.ownedBy = ownedBy;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("parent")
    public FolderMini parent;
    public WebLink withParent(FolderMini parent) {
        this.parent = parent;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("path_collection")
    public WebLinkPathCollection pathCollection;
    public WebLink withPathCollection(WebLinkPathCollection pathCollection) {
        this.pathCollection = pathCollection;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonSerialize(contentUsing = DateTimeSerializer.class)
    @JsonDeserialize(contentUsing = DateTimeDeserializer.class)
    @JsonProperty("purged_at")
    public OffsetDateTime purgedAt;
    public WebLink withPurgedAt(OffsetDateTime purgedAt) {
        this.purgedAt = purgedAt;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("sequence_id")
    public java.util.Map<String, Object> sequenceId;
    public WebLink withSequenceId(java.util.Map<String, Object> sequenceId) {
        this.sequenceId = sequenceId;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("shared_link")
    public WebLinkSharedLink sharedLink;
    public WebLink withSharedLink(WebLinkSharedLink sharedLink) {
        this.sharedLink = sharedLink;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonSerialize(contentUsing = DateTimeSerializer.class)
    @JsonDeserialize(contentUsing = DateTimeDeserializer.class)
    @JsonProperty("trashed_at")
    public OffsetDateTime trashedAt;
    public WebLink withTrashedAt(OffsetDateTime trashedAt) {
        this.trashedAt = trashedAt;
        return this;
    }
    @JsonProperty("type")
    public WebLinkTypeEnum type;
    public WebLink withType(WebLinkTypeEnum type) {
        this.type = type;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("url")
    public String url;
    public WebLink withUrl(String url) {
        this.url = url;
        return this;
    }
}

File: sdks/go/pkg/models/shared/weblink.go
package shared

import (
	"time"
)

type WebLinkItemStatusEnum string

const (
	WebLinkItemStatusEnumActive  WebLinkItemStatusEnum = "active"
	WebLinkItemStatusEnumTrashed WebLinkItemStatusEnum = "trashed"
	WebLinkItemStatusEnumDeleted WebLinkItemStatusEnum = "deleted"
)

// WebLinkPathCollection
// The tree of folders that this web link is contained in,
// starting at the root.
type WebLinkPathCollection struct {
	Entries    []FolderMini `json:"entries"`
	TotalCount int64        `json:"total_count"`
}

type WebLinkSharedLinkAccessEnum string

const (
	WebLinkSharedLinkAccessEnumOpen          WebLinkSharedLinkAccessEnum = "open"
	WebLinkSharedLinkAccessEnumCompany       WebLinkSharedLinkAccessEnum = "company"
	WebLinkSharedLinkAccessEnumCollaborators WebLinkSharedLinkAccessEnum = "collaborators"
)

type WebLinkSharedLinkEffectiveAccessEnum string

const (
	WebLinkSharedLinkEffectiveAccessEnumOpen          WebLinkSharedLinkEffectiveAccessEnum = "open"
	WebLinkSharedLinkEffectiveAccessEnumCompany       WebLinkSharedLinkEffectiveAccessEnum = "company"
	WebLinkSharedLinkEffectiveAccessEnumCollaborators WebLinkSharedLinkEffectiveAccessEnum = "collaborators"
)

type WebLinkSharedLinkEffectivePermissionEnum string

const (
	WebLinkSharedLinkEffectivePermissionEnumCanEdit     WebLinkSharedLinkEffectivePermissionEnum = "can_edit"
	WebLinkSharedLinkEffectivePermissionEnumCanDownload WebLinkSharedLinkEffectivePermissionEnum = "can_download"
	WebLinkSharedLinkEffectivePermissionEnumCanPreview  WebLinkSharedLinkEffectivePermissionEnum = "can_preview"
	WebLinkSharedLinkEffectivePermissionEnumNoAccess    WebLinkSharedLinkEffectivePermissionEnum = "no_access"
)

// WebLinkSharedLinkPermissions
// Defines if this link allows a user to preview, edit, and download an item.
// These permissions refer to the shared link only and
// do not supersede permissions applied to the item itself.
type WebLinkSharedLinkPermissions struct {
	CanDownload bool `json:"can_download"`
	CanEdit     bool `json:"can_edit"`
	CanPreview  bool `json:"can_preview"`
}

// WebLinkSharedLink
// The shared link object for this item. Will be
// `null` if no shared link has been created.
type WebLinkSharedLink struct {
	Access              *WebLinkSharedLinkAccessEnum             `json:"access,omitempty"`
	DownloadCount       int64                                    `json:"download_count"`
	DownloadURL         *string                                  `json:"download_url,omitempty"`
	EffectiveAccess     WebLinkSharedLinkEffectiveAccessEnum     `json:"effective_access"`
	EffectivePermission WebLinkSharedLinkEffectivePermissionEnum `json:"effective_permission"`
	IsPasswordEnabled   bool                                     `json:"is_password_enabled"`
	Permissions         *WebLinkSharedLinkPermissions            `json:"permissions,omitempty"`
	PreviewCount        int64                                    `json:"preview_count"`
	UnsharedAt          *time.Time                               `json:"unshared_at,omitempty"`
	URL                 string                                   `json:"url"`
	VanityName          *string                                  `json:"vanity_name,omitempty"`
	VanityURL           *string                                  `json:"vanity_url,omitempty"`
}

type WebLinkTypeEnum string

const (
	WebLinkTypeEnumWebLink WebLinkTypeEnum = "web_link"
)

// WebLink
// Web links are objects that point to URLs. These objects
// are also known as bookmarks within the Box web application.
//
// Web link objects are treated similarly to file objects,
// they will also support most actions that apply to regular files.
type WebLink struct {
	CreatedAt      *time.Time             `json:"created_at,omitempty"`
	CreatedBy      *UserMini              `json:"created_by,omitempty"`
	Description    *string                `json:"description,omitempty"`
	Etag           *string                `json:"etag,omitempty"`
	ID             string                 `json:"id"`
	ItemStatus     *WebLinkItemStatusEnum `json:"item_status,omitempty"`
	ModifiedAt     *time.Time             `json:"modified_at,omitempty"`
	ModifiedBy     *UserMini              `json:"modified_by,omitempty"`
	Name           *string                `json:"name,omitempty"`
	OwnedBy        *UserMini              `json:"owned_by,omitempty"`
	Parent         *FolderMini            `json:"parent,omitempty"`
	PathCollection *WebLinkPathCollection `json:"path_collection,omitempty"`
	PurgedAt       *time.Time             `json:"purged_at,omitempty"`
	SequenceID     map[string]interface{} `json:"sequence_id,omitempty"`
	SharedLink     *WebLinkSharedLink     `json:"shared_link,omitempty"`
	TrashedAt      *time.Time             `json:"trashed_at,omitempty"`
	Type           WebLinkTypeEnum        `json:"type"`
	URL            *string                `json:"url,omitempty"`
}
